File: BackEnd/src/CollectFlow.Domain/Entities/Customer.cs
namespace CollectFlow.Domain.Entities;

public class Customer : BaseEntity
{
    public Guid TenantId { get; set; }
    public Tenant Tenant { get; set; } = default!;

    public string Name { get; set; } = string.Empty;
    public string CompanyName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Notes { get; set; } = string.Empty;

    public ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
}

File: FrontEnd/src/types/customer.ts
export type Customer = {
  id: string;
  tenantId: string;
  name: string;
  companyName: string;
  email: string;
  phone: string;
  notes: string;
  createdAtUtc: string;
};

export type CreateCustomerRequest = {
  tenantId: string;
  name: string;
  companyName: string;
  email: string;
  phone: string;
  notes: string;
};
